rskj-core/src/main/java/org/ethereum/core/genesis/BlockChainLoader.java

```diff
@@ -121,20 +124,31 @@ public BlockChainImpl loadBlockchain() {
         Block bestBlock = blockStore.getBestBlock();
         if (bestBlock == null) {
             logger.info("DB is empty - adding Genesis");
-            for (RskAddress addr : genesis.getPremine().keySet()) {
-                repository.createAccount(addr);
-                InitialAddressState initialAddressState = genesis.getPremine().get(addr);
-                repository.addBalance(addr, initialAddressState.getAccountState().getBalance());
-                AccountState accountState = repository.getAccountState(addr);
-                accountState.setNonce(initialAddressState.getAccountState().getNonce());
-
-                if (initialAddressState.getContractDetails()!=null) {
-                    repository.updateContractDetails(addr, initialAddressState.getContractDetails());
-                    accountState.setStateRoot(initialAddressState.getAccountState().getStateRoot());
-                    accountState.setCodeHash(initialAddressState.getAccountState().getCodeHash());
```

Comment:
We should keep this line to set codeHash and keep consistency

Reply:
All this behaviour has been moved into the `GenesisLoader#mapFromJson` logic, the actual `AccountState` is created in the _loader_ with the proper _codeHash_ in place